src/Razor/src/Microsoft.VisualStudio.Editor.Razor/BackgroundParser.cs

```diff
@@ -185,7 +185,14 @@ public WorkParcel GetParcel()
                     }
                 }
 
-                _hasParcel?.Wait(_cancelSource.Token);
+                try
+                {
+                    _hasParcel?.Wait(_cancelSource.Token);
+                }
+                catch (Exception)
```

Comment:
Could we narrow this to a subset of `Exception`? I'm always spooped when I see `catch (Exception)`, but on the other hand I know chances of something else throwing an exception there are slim.

Reply:
In this case we're actually trying to take a hammer to it and smash out all exceptions because this code is in the legacy code base which started to exhibit some problems after making other pieces more performant.